includes/functions/disabled_upcoming.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * disabled-upcoming products functions
+ *
+ * @copyright 2018
+ * @license http://www.zen-cart.com/License/2_0.txt GNU Public License V2.0
+ * @author mc12345678
+ **/
+
+function zen_set_disabled_upcoming_status($products_id, $status) {
+    $sql = "UPDATE " . TABLE_PRODUCTS . "
+              SET products_status = " . (int)$status . ", products_date_available = '0001-01-01' WHERE products_id = " . (int)$products_id;
+
+    return $GLOBALS['db']->Execute($sql);
+}
+
+function zen_enable_disabled_upcoming() {
+
+    $date_range = time();
+
+    $zc_disabled_upcoming_date = date('Ymd', $date_range);
+
+    $disabled_upcoming_query = "SELECT products_id
+                                            FROM " . TABLE_PRODUCTS . "
+                                            WHERE products_status = 0
+                                            AND ((products_date_available <= " . $zc_disabled_upcoming_date . "
+                                            AND products_date_available != '0001-01-01'))
```

Comment:
Why the `((`double parentheses`))`?

Reply:
Held over from copy/paste from other areas where an `or` clause was present and needed to have double parentheses.  Not necessary to be present and serves no value.

Reply:
Shouldn't there be single-quotes around the first `products_date_available` AND-condition?

Reply:
Maybe I'm not folllowing.  `products_date_available` is a `datetime` database field.  Presumably such a field does not require quotes in order to reference the database content.  That field is being used as comparison against the formatted string date `$zc_disabled_upcoming_date`.  This is similar in construction as in includes/functions/specials.php, featured.php and salemaker.php (with below code taken from featured):
````php
    $featured_query = "select featured_id
                       from " . TABLE_FEATURED . "
                       where status = '0'
                       and (((featured_date_available <= " . $zc_featured_date . " and featured_date_available != '0001-01-01') and (expires_date > " . $zc_featured_date . "))
                       or ((featured_date_available <= " . $zc_featured_date . " and featured_date_available != '0001-01-01') and (expires_date = '0001-01-01'))
                       or (featured_date_available = '0001-01-01' and expires_date > " . $zc_featured_date . "))
                       ";
````

I did remove the single quotes around the value of `0` being compared against the field `products_status` as it is defined as a `tinyint` instead of a string. Is that perhaps the single quotes that appear to be missing? 

I *have* been pondering the MySQL 8.x issue(s) and whether a check against `IS NOT NULL` should be added in addition to the check against `!= '0001-01-01'`, but then also wondered if I was getting ahead of things with that.

Reply:
My bad (again); it just looked "funny" to have the quotes around the value when checking for equality but not when comparing.